common/src/main/kotlin/com/hartwig/actin/molecular/panel/PanelSpecifications.kt

```diff
@@ -19,10 +22,18 @@ class PanelSpecifications(panelSpecifications: Map<PanelTestSpecification, List<
         testSpec: PanelTestSpecification,
         negativeResults: Set<SequencedNegativeResult>?
     ): PanelTargetSpecification {
+        val unknownGenes = (negativeResults?.map(SequencedNegativeResult::gene)?.toSet() ?: emptySet()) - knownGenes
+        if (unknownGenes.isNotEmpty()) {
+            throw IllegalStateException(
+                "${logPanelName(testSpec)} has negative results associated containing " +
+                        "gene(s) not present in SERVE known genes: ${unknownGenes.joinToString()}." +
+                        "Correct this in the feed UI before continuing."
+            )
+        }
```

Comment:
I think it would be cleaner to only evaluate this when negative results exist, e.g.:

```suggestion
        negativeResults?.map(SequencedNegativeResult::gene)?.toSet()
            ?.let { it - knownGenes }.takeIf { it.isNotEmpty() }
            ?.let { unknownGenes ->
                throw IllegalStateException(
                    "${logPanelName(testSpec)} has negative results associated containing " +
                            "gene(s) not present in SERVE known genes: ${unknownGenes.joinToString()}." +
                            "Correct this in the feed UI before continuing."
                )
            }
```